docs/catalog/cloud/optimize-impact-on-customer-equipment.md

```diff
@@ -0,0 +1,46 @@
+---
+version: 1.0
+submitted_by: markus-ntt-seidl
+published_date: TBD
+category: cloud
+tags: 
+ - cloud
+ - role:cloud-engineer
+ - size:small
+---
+
+# Optimize impact on customer devices and equipment
+
+## Description
+
+Applications run on customer hardware or are displayed on it. The hardware used by the customer has a carbon footprint embodied through the production and electricity required while running. Optimising your software design and architecture to extend the life of the customer devices reduces the carbon intensity of the application.
+Ideally the customer can use the hardware until it's failure or until it becomes obsolete.
+
+
+## Solution
+
+Understand the devices and equipment your customers use to consume your services. Implement software patterns and architectures to minimize the need for customers to replace devices and upgrade equipment. For example, implement new features using code that is backwards compatible with older hardware and operating system versions, or manage the size of payloads so they don’t exceed the storage capacity of the target device.
```

Comment:
I understand the sentiment here - supporting older devices and operating systems by reducing the technical requirements to run your application - but it is not clear how that happens in this pattern description. It's not clear what changes can be applied or how I can read this pattern and apply this pattern to a system. If the goal is to extend the lifetime of customer devices, how would you know as the provider if that was successful?

Reply:
We are so high level with the patterns, is sometimes really difficult to get down on details. "Normal" Pattern books have a more focus in it which makes it easier (enterprise patterns or game architecture patterns or java patterns, ...)

How to support older hardware is pretty much application specific. If you have a phone you have to support older Android or iOS versions and also optimize UX (so it doesn't run slow). If you have a Video Chat Software client it would make sense to use hardware acceleration for video calls on older and newer hardware (looking at you Teams ;) ). In Games you support older GPUs (if that makes sense). 

The last point I do agree with you: "how would you know as the provider if that was successful?"
I guess in the end you can only guess, unless you keep statistics on used devices. But you'll never know if a user switched from a low-end device to a newer device just because of you or if the device died. 
Maybe you can market it as such? (runs on older hardware) Or get certifications for that?

Side Note:
I just checked and in Germany the "DE-UZ 215" is ready, that a certificate for Green Products https://www.blauer-engel.de/de/produktwelt/ressourcen-und-energieeffiziente-softwareprodukte . The criteria for that certificate states (roughly translated by me): "The software must be runnable on a reference system from the year of certificate application and on a reference system that is at least 5 years old"

Reply:
There is an english version, if you are interested: https://www.blauer-engel.de/en/productworld/resources-and-energy-efficient-software-products

Reply:
Fascinating and really cool find! I wonder if this pattern should morph to follow the DE-UZ 215 guidance and use it as a reference? I do like the concept of "Software must be able to run on X year old devices". Does that make sense?